float64 x
float64 y
time t
float64 x2
float64 y2
time t2